pkg/storage/localstore/subscription_pull_test.go

```diff
@@ -0,0 +1,693 @@
+			}
+			chunks[bin] = chunks[bin][1:]
+		}
+	}
+
+	// set a timeout on subscription
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	// signals that there were valid bins for this check to ensure test validity
+	var checkedBins int
+	// subscribe to every bin and validate returned values
+	for bin := uint8(0); bin <= uint8(chunk.MaxPO); bin++ {
+		// do not subscribe to bins that do not have chunks
+		if len(chunks[bin]) == 0 {
+			continue
+		}
+		// subscribe from start of the bin index
+		var since uint64
+		// subscribe until the first available chunk in bin,
+		// but not for it
+		until := chunks[bin][0].BinID - 1
+		if until <= 0 {
+			// ignore this bin if it has only one chunk left
+			continue
+		}
+		ch, stop := db.SubscribePull(ctx, bin, since, until)
+		defer stop()
+
+		// the returned channel should be closed
+		// because no chunks should be provided
+		select {
+		case d, ok := <-ch:
+			if !ok {
+				// this is expected for successful case
+				break
+			}
+			if d.BinID > until {
+				t.Errorf("got %v for bin %v, subscribed until bin id %v", d.BinID, bin, until)
+			}
+		case <-ctx.Done():
+			t.Error(ctx.Err())
+		}
+
+		// mark that the check is performed
+		checkedBins++
+	}
+	// check that test performed at least one validation
+	if checkedBins == 0 {
+		t.Fatal("test did not perform any checks")
+	}
+}
+
+// uploadRandomChunksBin uploads random chunks to database and adds them to
+// the map of addresses ber bin.
+func uploadRandomChunksBin(t *testing.T, db *DB, addrs map[uint8][]chunk.Address, addrsMu *sync.Mutex, wantedChunksCount *int, count int) {
+	addrsMu.Lock()
+	defer addrsMu.Unlock()
+
+	for i := 0; i < count; i++ {
+		ch := generateTestRandomChunk()
+
+		_, err := db.Put(context.Background(), chunk.ModePutUpload, ch)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		bin := db.po(ch.Address())
+		if _, ok := addrs[bin]; !ok {
+			addrs[bin] = make([]chunk.Address, 0)
+		}
+		addrs[bin] = append(addrs[bin], ch.Address())
+
+		*wantedChunksCount++
+	}
+}
+
+// readPullSubscriptionBin is a helper function that reads all chunk.Descriptors from a channel and
+// sends error to errChan, even if it is nil, to count the number of chunk.Descriptors
+// returned by the channel.
+func readPullSubscriptionBin(ctx context.Context, db *DB, bin uint8, ch <-chan chunk.Descriptor, addrs map[uint8][]chunk.Address, addrsMu *sync.Mutex, errChan chan error) {
+	var i int // address index
+	for {
+		select {
+		case got, ok := <-ch:
+			if !ok {
+				return
+			}
+			var err error
+			addrsMu.Lock()
+			if i+1 > len(addrs[bin]) {
+				err = fmt.Errorf("got more chunk addresses %v, then expected %v, for bin %v", i+1, len(addrs[bin]), bin)
+			} else {
+				addr := addrs[bin][i]
+				if !bytes.Equal(got.Address, addr) {
+					err = fmt.Errorf("got chunk bin id %v in bin %v %v, want %v", i, bin, got.Address.Hex(), addr.Hex())
+				} else {
+					var want shed.Item
+					var err1 error
```

Comment:
No need to have err1 variable defined and then explicitly set to err. I think that err can be set in the line below, just as @acud suggested in previous comment.